### PR TITLE
[BEAM-1140] delay stream termination after final mark to reduce test flakiness (stop gap measure).

### PR DESCRIPTION
**Pull request by tweise**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by davorbonaci** [comment]
R: @kennknowles

File: runners/apex/src/main/java/org/apache/beam/runners/apex/translation/operators/ApexReadUnboundedInputOperator.java [hunk 1]
```diff
@@ -106,6 +104,18 @@ private void emitWatermarkIfNecessary(long mark) {
 
   @Override
   public void endWindow() {
+    if (outputWatermark == GlobalWindow.TIMESTAMP_MAX_VALUE.getMillis()) {
```
Comment (kennknowles) [inline runners/apex/src/main/java/org/apache/beam/runners/apex/translation/operators/ApexReadUnboundedInputOperator.java]: This should really be `BoundedWindow.TIMESTAMP_MAX_VALUE`. It is inheritance on a static field that isn't really meant for inheritance (but, Java).

And even though this is fine, I would feel slightly better by using `>=` (aka `.isAfter`) which lets the reader know the idea and could be more lenient if some variation occurs accidentally.

**Comment by tweise** [comment]
Addressed comment.

File: runners/apex/src/main/java/org/apache/beam/runners/apex/translation/operators/ApexReadUnboundedInputOperator.java (hunk 1 above)
Comment (tweise) [inline runners/apex/src/main/java/org/apache/beam/runners/apex/translation/operators/ApexReadUnboundedInputOperator.java]: Done.

**Comment by kennknowles** [comment]
LGTM

**Pull request merged**